packages/esm-patient-common-lib/src/offline/patient.ts

```diff
@@ -0,0 +1,60 @@
+import { usePatient, getSynchronizationItems } from '@openmrs/esm-framework';
+import useSWR from 'swr';
+
+export const patientRegistrationSyncType = 'patient-registration';
+
+/**
+ * A hook similar to {@link usePatient} which also considers patients which have been newly registered
+ * in offline mode by the patient registration module.
+ */
+export function usePatientOrOfflineRegisteredPatient(patientUuid: string): ReturnType<typeof usePatient> {
```

Comment:
@manuelroemer Should we just make this what `usePatient` does?

Reply:
@brandones No, I don’t think so. That would only work if all usePatient users could correctly handle offline registered patients. Giving a module which doesn’t expect offline patients one would certainly break things, e.g. when a follow up API request is made with the patient‘s UUID. That would always fail because the patient doesn’t exist in the BE. But there are also other failure points.

Reply:
Makes sense, thanks @manuelroemer